src/compiler/checker.ts

```diff
@@ -16952,9 +16952,13 @@ namespace ts {
             if (operandType === silentNeverType) {
                 return silentNeverType;
             }
-            if (node.operator === SyntaxKind.MinusToken && node.operand.kind === SyntaxKind.NumericLiteral) {
+            const isOperandNumericLiteral = node.operand.kind === SyntaxKind.NumericLiteral;
+            if (isOperandNumericLiteral && node.operator === SyntaxKind.MinusToken) {
                 return getFreshTypeOfLiteralType(getLiteralType(-(<LiteralExpression>node.operand).text));
             }
+            if (isOperandNumericLiteral && node.operator === SyntaxKind.PlusToken) {

```

Comment:
Can you fold these into one check for `node.operand.kind === SyntaxKind.NumericLiteral`?